pkg/registry/customresourcedefinition/etcd.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package customresourcedefinition
+
+import (
+	"context"
+
+	"github.com/raushan2016/crd-apiserver/pkg/apis/apiextensions"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/generic"
+	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+// rest implements a RESTStorage for API services against etcd
+type REST struct {
+	*genericregistry.Store
+}
+
+// NewREST returns a RESTStorage object that will work against API services.
+func NewREST(scheme *runtime.Scheme, optsGetter generic.RESTOptionsGetter) *REST {
+	strategy := NewStrategy(scheme)
+
+	store := &genericregistry.Store{
+		NewFunc:                  func() runtime.Object { return &apiextensions.CustomResourceDefinition{} },
```

Comment:
>CustomResourceDefinition [](start = 74, length = 24)

are these needed ?

Reply:
Currently i am getting exception 


panic: storage codec doesn't seem to match given type: Internal type not encodable: apiextensions.CustomResourceDefinition is not suitable for converting to ["instance.raushank.io/v1beta1"] in scheme "pkg/runtime/scheme.go:101"

goroutine 1 [running]:
github.com/raushan2016/crd-apiserver/pkg/registry/customresourcedefinition.NewREST(0xc0002bc1c0, 0x1e66a00, 0xc0002827e0, 0xc0003745a0)
 /home/raushank/go/src/github.com/raushan2016/crd-apiserver/pkg/registry/customresourcedefinition/etcd.go:51 +0x41c
github.com/raushan2016/crd-apiserver/pkg/apiserver.completedConfig.New(0xc000543920, 0xc0007d0590, 0x1e675e0, 0xc00000e018)
 /home/raushank/go/src/github.com/raushan2016/crd-apiserver/pkg/apiserver/apiserver.go:107 +0x380
github.com/raushan2016/crd-apiserver/pkg/cmd/server.Run(0xc0002dd0e0, 0xc0002021e0, 0x1e675e0, 0xc00000e018)
 /home/raushank/go/src/github.com/raushan2016/crd-apiserver/pkg/cmd/server/server.go:60 +0x9a
github.com/raushan2016/crd-apiserver/pkg/cmd/server.NewServerCommand.func1(0xc000136280, 0xc000048680, 0x0, 0x8, 0x0, 0x0)
 /home/raushank/go/src/github.com/raushan2016/crd-apiserver/pkg/cmd/server/server.go:40 +0xa5
github.com/spf13/cobra.(*Command).execute(0xc000136280, 0xc00003a0a0, 0x8, 0x8, 0xc000136280, 0xc00003a0a0)
 /home/raushank/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x465
github.com/spf13/cobra.(*Command).ExecuteC(0xc000136280, 0xc00009c180, 0x1e675e0, 0xc00000e020)
 /home/raushank/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)

---
In reply to: [338368146](https://github.com/raushan2016/crd-apiserver/pull/1#discussion_r338368146) [](ancestors = 338368146)